File: backend/app/schemas/plant.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class PlantBase(BaseModel):
    name: str
    watering_interval_days: int | None = None
    next_watering_date: date | None = None
    last_watered_at: date | None = None
    notes: str = ""


class PlantCreate(PlantBase):
    # Client-generated UUID so creates are idempotent/offline-first (§11).
    id: uuid.UUID


class PlantRead(PlantBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    updated_at: datetime
    deleted_at: datetime | None = None

File: ios/PlantReviver/Models/Plant.swift
import Foundation
import SwiftData

/// Local source of truth for a plant. Carries the same syncable fields as the
/// backend (`id`, `updatedAt`, `deletedAt`) so offline-first sync can reconcile
/// (ARCHITECTURE.md §11).
@Model
final class Plant {
    @Attribute(.unique) var id: UUID
    var name: String
    var wateringIntervalDays: Int?
    var lastWateredAt: Date?
    var nextWateringDate: Date?
    var notes: String

    var createdAt: Date
    var updatedAt: Date
    var deletedAt: Date?

    init(
        id: UUID = UUID(),
        name: String,
        wateringIntervalDays: Int? = nil,
        notes: String = ""
    ) {
        self.id = id
        self.name = name
        self.wateringIntervalDays = wateringIntervalDays
        self.notes = notes
        let now = Date()
        self.createdAt = now
        self.updatedAt = now
    }
}
